torch/_inductor/fx_passes/pad_mm.py

```diff
@@ -504,6 +547,107 @@ def write_pad():
         return should_pad
 
 
+def get_context(
+    mat1,
+    mat2,
+    mat1_pre_padded,
+    mat2_pre_padded,
+    m_padded_length,
+    k_padded_length,
+    n_padded_length,
+):
+    context = AHContext()
+
+    context.add_feature("m", mat1.shape[0])
+    context.add_feature("k", mat1.shape[1])
+    context.add_feature("n", mat2.shape[1])
+
+    mat1_strides = mat1.stride()
+    mat2_strides = mat2.stride()
+    context.add_feature("mat1_stride_0", mat1_strides[0])
+    context.add_feature("mat1_stride_1", mat1_strides[1])
+    context.add_feature("mat2_stride_0", mat2_strides[0])
+    context.add_feature("mat2_stride_1", mat2_strides[1])
+
+    context.add_feature("m_padded_length", m_padded_length)
+    context.add_feature("k_padded_length", k_padded_length)
+    context.add_feature("n_padded_length", n_padded_length)
+
+    context.add_feature("mat1_iscontig", mat1.is_contiguous(), is_categorical=True)
+    context.add_feature("mat2_iscontig", mat2.is_contiguous(), is_categorical=True)
+
+    context.add_feature("mat1_align_size", get_alignment_size(mat1))
+    context.add_feature("mat2_align_size", get_alignment_size(mat2))
+
+    context.add_feature("mat1_dtype", mat1.dtype, is_categorical=True)
+    context.add_feature("mat2_dtype", mat2.dtype, is_categorical=True)
+
+    context.add_feature("prepadded_mat1", mat1_pre_padded, is_categorical=True)
+    context.add_feature("prepadded_mat2", mat2_pre_padded, is_categorical=True)
+
+    using_tf32 = "not_float_32"
+    if mat1.dtype == torch.float32:
+        using_tf32 = torch.backends.cuda.matmul.allow_tf32
+    context.add_feature("using_tf32", using_tf32, is_categorical=True)
+
+    return context
+
+
+def run_autoheuristic(
+    mat1,
+    mat2,
+    orig_bench_fn,
+    pad_bench_fn,
+    m_padded_length,
+    k_padded_length,
+    n_padded_length,
+    do_bench,
+    mat1_pre_padded,
+    mat2_pre_padded,
+):
+    def feedback_fn(choice):
+        if choice == orig_choice:
+            return do_bench(orig_bench_fn)
+        elif choice == pad_choice:
+            return do_bench(pad_bench_fn)
+        return None
+
+    def fallback():
+        return "autotune"
+
+    orig_choice = "orig"
+    pad_choice = "pad"
+    choices = [orig_choice, pad_choice]
+    feedback = LocalFeedback(feedback_fn)
+    context = get_context(
+        mat1,
+        mat2,
+        mat1_pre_padded,
+        mat2_pre_padded,
+        m_padded_length,
+        k_padded_length,
+        n_padded_length,
+    )
+    name = "pad_mm"
+    autoheuristic = AutoHeuristic(
+        fallback=fallback,
+        choices=choices,
+        feedback=feedback,
+        context=context,
+        name=name,
+        augment_context=pad_mm_operations(),
```

Comment:
How should `augment_context` be treated differently from just changing the context function? I understand generally what it's used for (adding new context), but I'm not sure how it's used in the context of this function.